### ISSUE TITLE
Improve configuration through declarative services

### ISSUE THREAD

**Issue by BryanHunt** [BryanHunt/mongo-emf#21]
Configuration of the Mongo instances should be possible through ConfigAdmin.  I propose restructuring the code to take better advantage of declarative services and service configuration.  IMongoDB will be refactored into IMongoFactory and the service will become a factory for instances of Mongo.

The changes should be structured such that users will still be able to use the code outside of an OSGi container.

**Comment by BryanHunt** [comment]
I have pushed an initial implementation of IMongoFactory to the services branch.

**Comment by BryanHunt** [comment]
I just pushed the latest changes to the mongo OSGi services.  All of the core bundles comple.  The unit tests still need work.

**Comment by BryanHunt** [comment]
I got the unit tests and example project working.  The URI scheme has changed from mongo:// to mongodb:// - this was necessary because of the way the Mongo is located by comparing the EMF URI to the MongoDB URI.

**Comment by BryanHunt** [comment]
I refactored all of the factories and converter service into OSGi services.  I have created a trial 0.7.0 p2 repository which can be found in the usual place.

**Comment by BryanHunt** [comment]
I have added metatype information for configuring IMongoProvider instances.  I also updated the 0.7.0 p2 repository.  The code has been merged into master.

**Comment by BryanHunt** [comment]
I refactored the resource set factory to use a generic resource set configurator (IResourceSetConfigurator) to build the ResourceSet.  The code that added the URI handlers to the resource set was moved into it's own resource set configurator.  This code has been pushed to master and is available in the 0.7.0 p2 repository.  I think this completes the major code changes.  I need to add a bit of documentation and I think the 0.7.0 release will be ready.
